test/nullable.jl

```diff
@@ -291,5 +299,111 @@ end
 @test Base.promote_op(-, Nullable{Float64}, Nullable{Int}) == Nullable{Float64}
 @test Base.promote_op(-, Nullable{DateTime}, Nullable{DateTime}) == Nullable{Base.Dates.Millisecond}
 
+# getindex
+@test_throws NullException Nullable()[]
+@test_throws NullException Nullable{Int}()[]
+@test Nullable(0)[] === 0
+
+@test_throws BoundsError Nullable()[-1]
+@test_throws BoundsError Nullable()[0]
+@test_throws BoundsError Nullable()[1]
+@test_throws BoundsError Nullable()[2]
+@test_throws BoundsError Nullable()[big(2)^64 + 1]
+@test_throws BoundsError Nullable(0)[-1]
+@test_throws BoundsError Nullable(0)[0]
+@test_throws BoundsError Nullable(0)[2]
+@test Nullable(0)[1] === 0
+
+# collect
+@test isempty(collect(Nullable())) && eltype(collect(Nullable())) == Union{}
```

Comment:
Split these `&&` into two lines, this is more readable and more precise about what fails.